metis-processor/src/main/java/eu/europeana/metis/processor/utilities/ImageEnhancerUtil.java

```diff
@@ -13,227 +16,338 @@
-            try {
-                if (hasThumbnailResolutionLowerThan400(webResourceWrapper)) {
-                    final String thumbnailHex = md5Hex(webResourceWrapper.getAbout());
-                    final String largeThumbnailObjectName = thumbnailHex + ThumbnailKind.LARGE.getNameSuffix();
-                    LOGGER.debug("{}\t=>\t{}", webResourceWrapper.getAbout(), largeThumbnailObjectName);
-                    byte[] largeThumbnail = s3Client.getIbmObject(largeThumbnailObjectName);
-                    if (ArrayUtils.isEmpty(largeThumbnail)) {
-                        largeThumbnail = s3Client.getAmazonObject(largeThumbnailObjectName);
-                    }
-                    if (isWebResourceResolutionAndImageResolutionSame(webResourceWrapper, largeThumbnail)) {
-//                        Files.write(Paths.get("/tmp/test_thumbnails/" + largeThumbnailObjectName), largeThumbnail);
-                        final long startTime = System.nanoTime();
-                        byte[] enhancedImage = enhanceImage(largeThumbnail, largeThumbnailObjectName);
-//                        Files.write(Paths.get("/tmp/test_thumbnails/enhanced-" + largeThumbnailObjectName), enhancedImage);
-                        final long elapsedTimeInNanoSec = System.nanoTime() - startTime;
-
-                        List<Thumbnail> thumbnails = generateThumbnails(webResourceWrapper, largeThumbnailObjectName, enhancedImage);
-                        uploadThumbnails(webResourceWrapper, thumbnails);
-                        appendSuccessReport(webResourceWrapper, enhancedImage, thumbnailHex, elapsedTimeInNanoSec, reportRow);
-                    } else {
-                        appendProcessedReport(webResourceWrapper, thumbnailHex, reportRow);
-                    }
-                }
-            } catch (MediaExtractionException | IOException e) {
-                LOGGER.error("Failed enhancing thumbnail image {} {}", webResourceWrapper.getAbout(), e);
-                appendFailReport(reportRow);
+  private void extractThumbnailEnhanceAndReUploadIt(RDF recordToProcess) {
+    List<Aggregation> curatedAggregationList = Optional.ofNullable(recordToProcess.getAggregationList())
+                                                       .orElse(Collections.emptyList());
+    final String recordId = recordToProcess.getProvidedCHOList().get(0).getAbout();
+    if (hasAggregationWithHasView(curatedAggregationList)) {
+      LOGGER.info("=>hasView");
+      curatedAggregationList.forEach(aggregationItem -> {
+            if (hasShownBy(aggregationItem)) {
+              LOGGER.info("=>shownByItem");
+              processShownBy(recordToProcess, aggregationItem, recordId);
             }
-        });
+            LOGGER.info("=>hasViewItem");
+            processHasView(recordToProcess, aggregationItem, recordId);
+          }
+      );
+    } else {
+      processSimpleEntry(recordToProcess, recordId);
```

Comment:
Is this valid here? The ticket says "Batch B consists of the records that DO have at least one hasView link". And this would process a record that does not have at least one hasView link?

Reply:
:thinking: true. removed.